### DISCUSSION TITLE
Ganache evm_mine lag

### DISCUSSION THREAD

**Discussion by ianrowan** [ConsenSys-archive/truffle#6048] [General Q&A] [answered]
Was curious how the `evm_mine` RPC command(on most recent version of Ganache which allows for params.blocks) resolves.

Basically we're running into intermittent issues in tests we're writing where the next line after awaiting the `evm_mine` RPC command isnt realizing the change in blocks sometimes, but adding almost any lag between almost guarantees the blocks have changed. For context we are using a 12 sec block time.

Main question is: Does `evm_mine` only submit a mine command, which is still mined at the normal blocktime, or should the new blocks be mined upon response from the RPC command?

**Comment by FlaviuWeb3** [comment, marked as answer]
The evm_mine RPC command in Ganache is used to manually mine a new block in the Ethereum Virtual Machine (EVM) simulation. When you send the evm_mine command via RPC, it does not immediately mine a block upon receiving the response. Instead, it adds a new block to the blockchain with a timestamp that corresponds to the current time.

In your case, where you're experiencing intermittent issues with the next line not recognizing the change in blocks, it's possible that the next line of code is executing before the new block is fully mined. The block time you mentioned is 12 seconds, which means it takes approximately 12 seconds for a new block to be added to the blockchain after calling evm_mine.

To ensure that the subsequent code recognizes the change in blocks consistently, you can introduce a delay after the evm_mine command. Adding a short delay, such as a few seconds, should be sufficient to allow the new block to be fully mined and added to the blockchain.

By introducing the delay, you give the EVM enough time to mine the block, and the subsequent code will see the updated state of the blockchain when it executes.
